### PR TITLE
add JSONSchema type

### PR DESCRIPTION
**Pull request by harunurhan**
This is just experimental for #252, I changed only single file to use `JSONSchema` as type for `schema`.

It seems more clean and typing is more strict of course, what do you think? @jmartinm @zzacharo

### CODE REVIEW

File: src/shared/interfaces/json-schema.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import { SchemaOption } from './schema-option';
+
+export interface JSONSchema extends SchemaOption {
```
Comment (jmartinm) [inline src/shared/interfaces/json-schema.ts]: Where did you get the list from? In case you missed some key, there is this example https://gist.github.com/enriched/c84a2a99f886654149908091a3183e15

Do we need to add $ref ?

File: src/object-field/object-field.component.html [hunk 2]
```diff
@@ -21,7 +21,7 @@
   <!-- ADD SUB FIELD FROM SCHEMA DROPDOWN -->
   <tr>
     <td class="button-holder">
-      <add-field-dropdown [fields]="keys" [pathString]="pathString" (onFieldAdd)="onFieldAdd($event)" [schema]="schema.properties">+</add-field-dropdown>
```
Comment (jmartinm) [inline src/object-field/object-field.component.html]: Why is this change needed?

File: src/shared/interfaces/json-schema.ts (hunk 1 above)
Comment (harunurhan) [inline src/shared/interfaces/json-schema.ts]: I got it somewhere like this link but, simplified it. My idea is to add the keys as we need/support them to keep it simple.

File: src/object-field/object-field.component.html (hunk 2 above)
Comment (harunurhan) [inline src/object-field/object-field.component.html]: I also unified the `@Input() schema`, `add-field-dropdown` was taking `schema.properties` as input and that might be a bit confusing later, it's better to be able to assume that when you see `schema`, it's JSONSchema 😄

**Comment by jmartinm** [comment]
I think this is a good idea. If you can fix the conflicts we can go ahead and merge it.
